example-site/components/TableErrorValidation/TableWithValidation.tsx

```diff
@@ -0,0 +1,373 @@
+
+			{pageErrors.length >= 2 && (
+				<PageAlert
+					focusOnUpdate={pageErrors}
+					title="There is a problem"
+					tone="error"
+				>
+					<Text as="p">Please correct the following fields and try again</Text>
+					<UnorderedList>
+						{pageErrors.map(({ href, message }) => {
+							return (
+								<ListItem key={href}>
+									<TextLink href={href}>{message}</TextLink>
+								</ListItem>
+							);
+						})}
+					</UnorderedList>
+				</PageAlert>
+			)}
+
+			<TextInput
+				id={locationTextInputId}
+				invalid={pageErrors.some(({ name }) => name === 'location')}
+				label="Location of assessment"
+				message={pageErrors.find(({ name }) => name === 'location')?.message}
+				onChange={(event: ChangeEvent<HTMLInputElement>) =>
+					setLocationValue(event.target.value)
+				}
+				name="location"
+				required
+				value={locationValue}
+			/>
+
+			<DatePicker
+				id={dateOfAssessmentDatePickerId}
+				invalid={pageErrors.some(({ name }) => name === 'date')}
+				label="Date of assessment"
+				message={pageErrors.find(({ name }) => name === 'date')?.message}
+				name="date"
+				onChange={setDateValue}
+				required
+				value={dateValue}
+			/>
+
+			<Stack gap={1}>
+				<H2>Assessment files</H2>
+
+				<Box css={tableErrors.length ? undefined : { display: 'none' }}>
+					<SectionAlert
+						id="section-alert"
+						ref={tableSectionAlertRef}
+						tabIndex={-1}
+						title="You must provide decisions for each change in the table below"
+						tone="error"
+					>
+						<Stack gap={0.5} alignItems="flex-start">
+							<UnorderedList>
+								{tableErrors.map(({ href, message }) => (
+									<ListItem key={href}>
+										<TextLink href={href}>{message}</TextLink>
+									</ListItem>
+								))}
+							</UnorderedList>
+						</Stack>
+					</SectionAlert>
+				</Box>
+
+				<DataTable
+					assessmentFiles={assessmentFiles}
+					ref={tableRef}
+					headingId={tableHeadingId}
+					tableId={tableId}
+					openDrawer={(assessmentFile: AssessmentFile) => {
+						openDrawer();
+						setCurrentAssessmentFile(assessmentFile);
+					}}
+				/>
+			</Stack>
+
+			<ButtonGroup>
+				<Button onClick={onSaveAndContinue}>Save and continue</Button>
+				<Button variant="secondary">Save and exit</Button>
+				<Button variant="text">Cancel</Button>
+			</ButtonGroup>
+
+			<Drawer
+				isOpen={isDrawerOpen}
+				onClose={closeDrawer}
+				title="Medium Drawer"
+				actions={
+					<ButtonGroup>
+						<Button onClick={closeDrawer}>Done</Button>
+					</ButtonGroup>
+				}
+				elementToFocusOnClose={
+					tableErrors.length ? tableSectionAlertRef.current : undefined
+				}
+			>
+				<Stack gap={2}>
+					<Text>{currentAssessmentFile?.documentType}</Text>
+
+					<Button onClick={mockUploadFile}>Mock upload file</Button>
+
+					<Button onClick={mockUploadError}>Mock upload error</Button>
+
+					<Button variant="tertiary" onClick={removeFile}>
+						Remove file
+					</Button>
```

Comment:
This shouldn't exist for a file that hasn't been uploaded yet

Reply:
- [ ] conditionally show "Remove file" button